cmake/CompileBoost.cmake

```diff
@@ -163,16 +178,16 @@ if(WIN32)
   # properly for config mode. So we use the old way on Windows
   #  find_package(Boost 1.72.0 EXACT QUIET REQUIRED CONFIG PATHS ${BOOST_HINT_PATHS})
   # I think depending on the cmake version this will cause weird warnings
-  find_package(Boost 1.78 COMPONENTS filesystem iostreams serialization system program_options)
+  find_package(Boost 1.86 COMPONENTS filesystem iostreams serialization system program_options url)
```

Comment:
There are lots of places where the boost version is defined. It's a time to define it in one place and use everywhere.